src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      a <= 0 || b <= 0 || c <= 0 || a >= (b + c) || b >= (a + c) || c >= (b + a)
+    ) {
+      throw new Error('One of side length less than or equal to zero'
+        + 'or the longest side of a triangle is >= than a sum of two others');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
```

Comment:
Use `this` destructuring, cause you use` this.a`, `this.b`, `this.c` in several places